packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -106,42 +73,59 @@ abstract class Scheduler extends BindingBase {
     ui.window.onBeginFrame = handleBeginFrame;
   }
 
-  static Scheduler _instance;
+  /// The current [Scheduler], if one has been created.
   static Scheduler get instance => _instance;
+  static Scheduler _instance;
 
+
+  /// The strategy to use when deciding whether to run a task or not.
+  /// 
+  /// Defaults to [defaultSchedulingStrategy].
   SchedulingStrategy schedulingStrategy = defaultSchedulingStrategy;
 
   static int _taskSorter (_TaskEntry e1, _TaskEntry e2) {
-    // Note that we inverse the priority.
     return -e1.priority.compareTo(e2.priority);
   }
   final PriorityQueue<_TaskEntry> _taskQueue = new HeapPriorityQueue<_TaskEntry>(_taskSorter);
 
-  /// Whether this scheduler already requested to be called from the event loop.
-  bool _hasRequestedAnEventLoopCallback = false;
-
-  /// Whether this scheduler already requested to be called at the beginning of
-  /// the next frame.
-  bool _hasRequestedABeginFrameCallback = false;
-
-  /// Schedules the given [task] with the given [priority].
+  /// Schedules the given `task` with the given `priority`.
+  ///
+  /// Tasks will be executed between frames, in priority order,
```

Comment:
I've added something saying the tasks should be short. I haven't said exactly how many we'll execute since this really needs to be reworked some.